src/cmap/connection_pool_events.ts

```diff
@@ -8,11 +21,23 @@ import type { ConnectionPool, ConnectionPoolOptions } from './connection_pool';
  * @public
  * @category Event
  */
-export class ConnectionPoolMonitoringEvent {
+export abstract class ConnectionPoolMonitoringEvent {
```

Comment:
I don't think we can make this class abstract, since it's public.

Reply:
From what I understand, this class has a private constructor, so users shouldn't be constructing these in their own code since constructing this event (and any of the other subclasses for that matter) isn't a part of our public API. As such, making this abstract would only break things for users that are _already_ misusing our API.